symplyphysics/laws/electricity/transmission_lines/transmission_matrix_of_π_type_circuit.py

```diff
@@ -0,0 +1,100 @@
+    dimensionless,
+    convert_to,
+)
+from symplyphysics.core.dimensions import assert_equivalent_dimension
+from symplyphysics.core.expr_comparisons import expr_equals
+from symplyphysics.laws.electricity.transmission_lines import transmission_matrix_for_a_series_load_in_line as series_law
+from symplyphysics.laws.electricity.transmission_lines import transmission_matrix_for_a_parallel_load_in_line as parallel_law
+
+## Description
+## The transmission parameters matrix is one of the ways to describe a microwave device. The ABCD-parameters of the device act as elements
+## of this matrix. The matrix equation relates the input voltage and input current to the output voltage and output current.
+## The π-type circuit consists of the first impedance connected in parallel, the second impedance connected in series, and the
+## third impedance connected in parallel.
+## Knowing impedances, it is possible to calculate the parameters A, B, C, D of the transmission matrix of this line.
+
+## Law is: Matrix([[A, B], [C, D]]) = Matrix([[1 + Z3 / Z2, Z3], [(1 / Z1) + (1 / Z2) + Z3 / (Z1 * Z2), 1 + Z3 / Z1]]), where
+## A - ratio of input voltage to output voltage at idle at the output,
+## B - ratio of input voltage to output current in case of a short circuit at the output,
+## C - ratio of input current to output voltage at idle at the output,
+## D - ratio of input current to output current in case of a short circuit at the output,
+## Z1 - first impedance,
+## Z2 - second impedance,
+## Z3 - third impedance.
+
+parameter_voltage_to_voltage = Symbol("parameter_voltage_to_voltage", dimensionless)
+parameter_impedance = Symbol("parameter_impedance", units.impedance)
+parameter_conductance = Symbol("parameter_conductance", units.conductance)
+parameter_current_to_current = Symbol("parameter_current_to_current", dimensionless)
+
+first_impedance = Symbol("first_impedance", units.impedance)
+second_impedance = Symbol("second_impedance", units.impedance)
+third_impedance = Symbol("third_impedance", units.impedance)
+
+
+law = Eq(Matrix([[parameter_voltage_to_voltage, parameter_impedance], [parameter_conductance, parameter_current_to_current]]),
+         Matrix([[1 + third_impedance / second_impedance, third_impedance], [(1 / first_impedance) + (1 / second_impedance) + third_impedance / (first_impedance * second_impedance), 1 + third_impedance / first_impedance]]))
+
+# This law might be derived via "transmission_matrix_for_a_series_load_in_line" law and
+# "transmission_matrix_for_a_parallel_load_in_line" law.
+
+parallel_law_applied_1 = parallel_law.law.subs({
+    parallel_law.load_impedance: first_impedance,
+})
+matrix_derived_1 = solve(parallel_law_applied_1, [parallel_law.parameter_voltage_to_voltage, parallel_law.parameter_impedance, parallel_law.parameter_conductance, parallel_law.parameter_current_to_current], dict=True)[0]
+matrix_derived_1[parallel_law.parameter_impedance] = matrix_derived_1[parallel_law.parameter_impedance].scale_factor
```

Comment:
You can use `0` instead of `Quantity` in the `parallel_law` to avoid this trick. Same for `series_law`

Reply:
> You can use `0` instead of `Quantity` in the `parallel_law` to avoid this trick. Same for `series_law`

1) Do you mean to do it like this?
parallel_law_applied_1 = parallel_law.law.subs({
    parallel_law.load_impedance: first_impedance,
    parallel_law.parameter_impedance: 0,
})
If you do this, solve() returns an empty list.

2) Or do you mean that in the laws "transmission_matrix_for_a_series_load_in_line" and "transmission_matrix_for_a_parallel_load_in_line" you need to replace zero values with 0? (see screenshot)
![image](https://github.com/blackyblack/symplyphysics/assets/100017508/7aafac52-b847-4a95-ab2c-993a04d68d17)

Reply:
2 option

Reply:
> 2\. transmission_matrix_for_a_series_load_in_line

But then the parameter "B"(ohm) returned by the law "transmission_matrix_for_a_parallel_load_in_line" and the parameter "C"(Siemens) returned by the law "transmission_matrix_for_a_series_load_in_line" will have no dimension, which is incorrect.

Reply:
They are zero, and we have a convention to let zero be any dimension.